src/index.ts

```diff
@@ -1,28 +1,25 @@
-import * as ended from 'is-stream-ended';
 import {Transform} from 'stream';
 
 /**
  * Push an array of items into a Transform stream.
  * @param array The array you want to push to the stream.
  * @param stream The Transform stream into which array items are pushed.
  */
-export async function split(array: Array<{}>, stream: Transform) {
-  return new Promise<boolean>((resolve, reject) => {
-    const arr = [].slice.call(array);
-    function loopyloop() {
-      // Ensure the stream wasn't closed by the consumer.
-      const isEnded = ended(stream);
-      // Ensure al items from the array haven't been pushed.
-      const cont = !isEnded && arr.length > 0;
-      if (cont) {
-        stream.push(arr.shift());
-        // For large arrays, use setImmediate to ensure other microtasks
-        // and I/O operations have a chance to execute.
-        setImmediate(loopyloop);
-      } else {
-        resolve(isEnded);
+export async function split(
+    array: Array<{}>, stream: Transform): Promise<boolean> {
+  const arr = [].slice.call(array);
+  let ended = false;
+
+  for (const item of arr) {
+    ended = !stream.writable || !stream.readable;
```

Comment:
That's all?

Reply:
Initially I started seeing a bunch of write after end errors, I traced it down to `is-stream-ended` only checking the readable state of a stream. Despite `end()` being a writable method, I think it will propagate to the readable state as well and I think previously `setImmediate` gave us time to let the readable state get updated. 

In my quest to find out if the stream has ended without explicitly checking the `readableState` or `writableState` (trying to appease TypeScript here) I came across these properties, which appear to only get set to false when the stream has ended.

Reply:
Would you be able to put it in is-stream-ended? I had to play with a lot of options in that module. One fix would break something else, over and over. Not saying I landed at the perfect solution, but if this works, patching it upstream could have benefits as well.